datadog/resource_datadog_integration.go

```diff
@@ -0,0 +1,171 @@
+	client := meta.(*datadog.Client)
+
+	if _, err := client.GetIntegrationPD(); err != nil {
+		if strings.Contains(err.Error(), "404 Not Found") {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+func buildIntegrationPagerduty(d *schema.ResourceData) (*datadog.IntegrationPDRequest, error) {
+	pd := &datadog.IntegrationPDRequest{}
+	pd.SetSubdomain(d.Get("subdomain").(string))
+	pd.SetAPIToken(d.Get("api_token").(string))
+
+	schedules := []string{}
+	for _, s := range d.Get("schedules").([]interface{}) {
+		schedules = append(schedules, s.(string))
+	}
+	pd.Schedules = schedules
+
+	services := []datadog.ServicePDRequest{}
+	for _, sInterface := range d.Get("services").([]interface{}) {
+		s := sInterface.(map[string]interface{})
+
+		service := datadog.ServicePDRequest{}
+		service.SetServiceName(s["service_name"].(string))
+		service.SetServiceKey(s["service_key"].(string))
+
+		services = append(services, service)
+	}
+	pd.Services = services
+
+	return pd, nil
+}
+
+func resourceDatadogIntegrationPagerdutyCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*datadog.Client)
+
+	pd, err := buildIntegrationPagerduty(d)
+	if err != nil {
+		return fmt.Errorf("Failed to parse resource configuration: %s", err.Error())
+	}
+
+	if err := client.CreateIntegrationPD(pd); err != nil {
+		return fmt.Errorf("Failed to create integration pagerduty using Datadog API: %s", err.Error())
+	}
+
+	pdIntegration, err := client.GetIntegrationPD()
+	if err != nil {
+		return fmt.Errorf("error retrieving integration pagerduty: %s", err.Error())
+	}
+
+	d.SetId(pdIntegration.GetSubdomain())
+
+	return nil
+}
+
+func resourceDatadogIntegrationPagerdutyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*datadog.Client)
+
+	pd, err := client.GetIntegrationPD()
+	if err != nil {
+		return err
+	}
+
+	services := []map[string]string{}
+	for _, service := range pd.Services {
+		services = append(services, map[string]string{
+			"service_name": service.GetServiceName(),
+			"service_key":  service.GetServiceKey(),
+		})
+	}
+
+	d.Set("services", services)
+	d.Set("subdomain", pd.GetSubdomain())
+	d.Set("schedules", pd.Schedules)
+	d.Set("api_token", pd.GetAPIToken())
+
+	return nil
+}
+
+func resourceDatadogIntegrationPagerdutyUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*datadog.Client)
+
+	pd, err := buildIntegrationPagerduty(d)
+	if err != nil {
+		return fmt.Errorf("Failed to parse resource configuration: %s", err.Error())
+	}
+
+	if err := client.UpdateIntegrationPD(pd); err != nil {
+		return fmt.Errorf("Failed to create integration pagerduty using Datadog API: %s", err.Error())
+	}
+
+	return resourceDatadogIntegrationPagerdutyRead(d, meta)
+}
+
+func resourceDatadogIntegrationPagerdutyDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*datadog.Client)
+
+	return client.DeleteIntegrationPD()
```

Comment:
Depending on how expressive the errors are in the DD client library, you may want to wrap this error. A lot of terraform code uses github.com/hashicorp/errwrap for this, but generally I just use `return fmt.Errorf("Error while deleting integration: %v", err)`